Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -383,7 +383,14 @@ extension NSManagedObjectContext: MessageDatabaseSession {
             dto.pinnedBy = try saveUser(payload: pinnedByUser)
         }
 
-        dto.quotedMessage = try payload.quotedMessage.flatMap { try saveMessage(payload: $0, for: cid) }
+        if let quotedMessage = payload.quotedMessage {
+            dto.quotedMessage = try saveMessage(payload: quotedMessage, for: cid)
+        } else if let quotedMessageId = payload.quotedMessageId {
+            // In case we do not have a fully formed quoted message in the payload, we check for quotedMessageId. This can happen in the case of nested quoted messages.
```

Comment:
The formatter doesn't automatically wrap comments. I recommend setting the Xcode guide to 132 to make sure also the comments are wrapped at the correct column.

<img width="793" alt="Screen Shot 2021-07-13 at 08 58 00" src="https://user-images.githubusercontent.com/6388510/125405826-6f8e0c80-e3b8-11eb-9b11-19ce941052ce.png">

Reply:
Thanks for this tip!